workflow/common/util.go

```diff
@@ -320,10 +393,12 @@ func substituteParams(tmpl *wfv1.Template, globalParams, localParams map[string]
 	// Now replace the rest of substitutions (the ones that can be made) in the template
 	replaceMap = make(map[string]string)
 	for _, inParam := range globalReplacedTmpl.Inputs.Parameters {
-		if inParam.Value == nil {
-			return nil, errors.InternalErrorf("inputs.parameters.%s had no value", inParam.Name)
+		if inParam.Regexp == "" { //this can still be set if we're handling a WorkflowTemplate
```

Comment:
If we do https://github.com/argoproj/argo/pull/1490/files#r308056723, inParam here won't have `Regexp`.